src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -129,21 +136,129 @@ class CreateTableStatementSegment(
         ),
         Ref("DatatypeSegment"),
     ),
+    LocalVariableNameSegment=RegexParser(
+        r"`?[a-zA-Z0-9_]*`?",
+        CodeSegment,
+        name="declared_variable",
+        type="variable",
+    ),
+    SessionVariableNameSegment=RegexParser(
+        r"[@][a-zA-Z0-9_]*",
+        CodeSegment,
+        name="declared_variable",
+        type="variable",
+    ),
 )
 
 mysql_dialect.replace(
     DelimiterSegment=OneOf(Ref("SemicolonSegment"), Ref("TildeSegment")),
     TildeSegment=StringParser(
         "~", SymbolSegment, name="tilde", type="statement_terminator"
     ),
+    ParameterNameSegment=RegexParser(
+        r"`?[A-Za-z0-9_]*`?", CodeSegment, name="parameter", type="parameter"
+    ),
+)
+
+mysql_dialect.patch_lexer_matchers(
+    [
+        RegexLexer(
+            "ampersand",
+            r"[@][a-zA-Z0-9_]*",
+            CodeSegment,
+        ),
```

Comment:
Why `ampersand`? An ampersand is an `&` not a `@`. I believe the latter is called an `at sign`: https://en.wikipedia.org/wiki/List_of_typographical_symbols_and_punctuation_marks

Reply:
@alanmcruickshank  this was a good one, I am not sure why my brain thought that it was an ampersand...lol. I renamed this, and because `atsign` is not an existing lexer_matcher I had to change the method to be `insert_lexer_matchers` instead.  All good now.